iam.tf

```diff
@@ -0,0 +1,52 @@
+locals {
+  lambda_basic_execution_role_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name = "iam_for_${var.environment}_${var.file_name}"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : "sts:AssumeRole",
+        "Principal" : {
+          "Service" : "lambda.amazonaws.com"
+        },
+        "Effect" : "Allow",
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "logs" {
+  name = "${var.environment}_${var.file_name}_logs"
+  role = aws_iam_role.iam_for_lambda.name
+  policy = jsonencode({
+    "Statement" : [
+      {
+        "Action" : [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents",
+        ],
+        "Effect" : "Allow",
+        "Resource" : "arn:aws:logs:${data.aws_region.current.name}:521380623667:*",
```

Comment:
construct `arn` as local?

And Is there a variable here?

Reply:
done, constructed it as local and aws_region as data source is used to construct the arn